### DISCUSSION TITLE
What does "No -c option found" mean?

### DISCUSSION THREAD

**Discussion by ghost** [ccache/ccache#244] [Questions & answers] [answered]
Hello,
I just would like to know, does the following is normal or an issue:
```
[2018-03-22T13:25:40.417552 14845] No -c option found
[2018-03-22T13:25:40.417575 14845] Failed; falling back to running the real compiler
```
I think it causing no so good `hit (direct)` result.
I can give more information and even the whole log file if needed.
Thanks!

**CCACHE 3.4.2-13 git5dc502e**

**Comment by vaneyli** [comment]
You should show just one cycle log of ccache  not the whole log.    cat log|grep -rn "14845]"

**Comment by afbjorklund** [comment, marked as answer]
@cranes-bill : that is normal, for simple cases - since ccache doesn't cache any linking...
You should do your compilation in one step, and then linking in the next. Like with `make` ?

``` console
$ cat Makefile 
hello: hello.o
$ make
cc    -c -o hello.o hello.c
cc   hello.o   -o hello
```
```
[2018-03-30T13:04:22.041689 15621] Result: cache hit (direct)
[2018-03-30T13:04:22.043905 15622] Result: called for link
```

instead of just:
cc     hello.c   -o hello

This is also scales much better for more real-life projects, and with multiple jobs.
You can do incremental compilation (`make`), and parallell complilation (`make -j`).

``` console
$ make
make: 'hello' is up to date.
```

**Comment by ghost** [comment]
Aha, clear!
Thank you very much!
